beets/importer.py

```diff
@@ -546,10 +546,22 @@ def _emit_imported(self, lib):
             return
         plugins.send('album_imported', lib=lib, album=self.album)
 
-    def emit_created(self, session):
-        """Send the `import_task_created` event for this task.
+    def handle_created(self, session):
+        """Send the `import_task_created` event for this task. Return a list of
+        tasks that should continue through the pipeline. By default, this is a
+        list containing only the task itself, but plugins can replace the task
+        with new ones.
         """
-        plugins.send('import_task_created', session=session, task=self)
+        tasks = plugins.send('import_task_created', session=session, task=self)
+        if not tasks:
+            tasks = [self]
+        else:
+            # The plugins gave us a list of lists of task. Flatten it.
+            flat_tasks = []
+            for inner in tasks:
+                flat_tasks += inner
+            tasks = [t for t in flat_tasks if t]
```

Comment:
Is this step necessary? Are we expecting some of the tasks to be `None`?

If not, we could also do something clever like `list(itertools.chain.from_iterable(flat_tasks))` or `[t for ts in flat_tasks for t in ts]`, just for fun. :smiley: